[AOTInductor] Modify test for Memory tracking for memory-related

muchulee8 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #150269

operations

Summary:
Fix the test for memory tracking. This PR does:
(1) Add tracking before and after for all memory-related operations.
Make sure the operation do indeed captures memory both in CUDA and
torch's CUDACachAllocator Make sure the operation do indeed captures
consumed memory both in CUDA and torch's CUDACachAllocator.
(2) Keep track of memory being reserved by CUDACacheAllocator in
torch and it's relationship with global CUDA memory consumption.

Test Plan:
This PR is adding tests.

Reviewers:

Subscribers:

Tasks:

Tags:

muchulee8 committed:
[AOTInductor] Modify test for Memory tracking for memory-related
operations

Summary:
Fix the test for memory tracking. This PR does:
(1) Add tracking before and after for all memory-related operations.
Make sure the operation do indeed captures memory both in CUDA and
torch's CUDACachAllocator Make sure the operation do indeed captures
consumed memory both in CUDA and torch's CUDACachAllocator.
(2) Keep track of memory being reserved by CUDACacheAllocator in
torch and it's relationship with global CUDA memory consumption.

Test Plan:
This PR is adding tests.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

desertfire commented on test/cpp/aoti_inference/test.cpp:
```diff
@@ -511,79 +514,107 @@ void test_aoti_free_buffer(bool use_runtime_constant_folding) {
 
   // Swap and free #2, no data should exist in memory now.
-  // However, the folded constants still occupies the CUDA memory in
+  // However, the folded constants might still occupies the CUDA memory in
```
What does might mean here, and does it make any of the following ASSERT_EQ non-deterministic?

muchulee8 replied:
It's actually the opposite, this PR's purpose is to make it from non-deterministic to deterministic.

Previously, we did things like (take the example immediate below): `ASSERT_EQ(initMemory + DATASIZE - 2 * FOLDEDDATASIZE, updateMemory1);`
We changed it to `ASSERT_EQ(initMemory + DATASIZE - (torchReserved2 - initTorchReserved), updateMemory1);` 

When we say "still occupies the CUDA memory", we are making the assumption that CUDACacheAllocator will not do any allocation/free between the free. I changed this to "might", because it's possible that CUDACacheAllocator has freed it's memory space along side with the free of folded-const's `at::Tensor`, thus our change of replacing `FOLDEDDATASIZE` with `(torchReserved2 - initTorchReserved)`, so that we are sure that the whole CUDA memory taken is tracked, and whether or not the folded constants are correctly freed are tracked by active memory.

muchulee8 commented:
@pytorchbot merge

muchulee8 commented:
@pytorchbot merge -r main

muchulee8 committed:
Update

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge -f "Unrelated build"

muchulee8 committed:
Update on "[AOTInductor] Modify test for Memory tracking for memory-related"


operations

Summary:
Fix the test for memory tracking. This PR does:
(1) Add tracking before and after for all memory-related operations.
Make sure the operation do indeed captures memory both in CUDA and
torch's CUDACachAllocator Make sure the operation do indeed captures
consumed memory both in CUDA and torch's CUDACachAllocator.
(2) Keep track of memory being reserved by CUDACacheAllocator in
torch and it's relationship with global CUDA memory consumption.

Test Plan:
This PR is adding tests.

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

muchulee8 commented:
@pytorchbot merge -f "Unrelated errors"
